### ISSUE TITLE
Support 0bda:818b wifi dongle with rtlwifi/rtl8192eu_nic.bin

### ISSUE THREAD

**Issue by pepijndevos** [ev3dev/ev3dev#1115]
### What are you trying to do?

I'm trying to connect a cheapo "Maxxter" WiFi dongle to my EV3.

### What did you expect to happen?
When connecting the device on my Ubuntu 18.04 laptop I see the following:

```
[16064.046320] usb 2-1.2: new high-speed USB device number 8 using ehci-pci
[16064.159672] usb 2-1.2: New USB device found, idVendor=0bda, idProduct=818b
[16064.159680] usb 2-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[16064.159684] usb 2-1.2: Product: 802.11n NIC
[16064.159689] usb 2-1.2: Manufacturer: Realtek
[16064.159692] usb 2-1.2: SerialNumber: 00e04c000001
[16064.324664] usb 2-1.2: Vendor: Realtek
[16064.324670] usb 2-1.2: Product: 802.11n NIC
[16064.324673] usb 2-1.2: Serial not available.
[16064.324678] usb 2-1.2: rtl8192eu_parse_efuse: dumping efuse (0x200 bytes):
[16064.324682] usb 2-1.2: 00: 29 81 00 7c 01 40 03 00
[16064.324685] usb 2-1.2: 08: 40 74 04 50 14 00 00 00
[16064.324689] usb 2-1.2: 10: 23 26 27 27 28 28 20 23
[16064.324692] usb 2-1.2: 18: 24 24 25 24 ef ef ff ff
[16064.324696] usb 2-1.2: 20: ff ff ff ff ff ff ff ff
[16064.324699] usb 2-1.2: 28: ff ff ff ff ff ff ff ff
[16064.324703] usb 2-1.2: 30: ff ff ff ff ff ff ff ff
[16064.324706] usb 2-1.2: 38: ff ff 28 26 26 24 24 24
[16064.324709] usb 2-1.2: 40: 25 23 23 21 21 24 ef ef
[16064.324713] usb 2-1.2: 48: ff ff ff ff ff ff ff ff
[16064.324716] usb 2-1.2: 50: ff ff ff ff ff ff ff ff
[16064.324720] usb 2-1.2: 58: ff ff ff ff ff ff ff ff
[16064.324723] usb 2-1.2: 60: ff ff ff ff ff ff ff ff
[16064.324726] usb 2-1.2: 68: ff ff ff ff ff ff ff ff
[16064.324730] usb 2-1.2: 70: ff ff ff ff ff ff ff ff
[16064.324733] usb 2-1.2: 78: ff ff ff ff ff ff ff ff
[16064.324737] usb 2-1.2: 80: ff ff ff ff ff ff ff ff
[16064.324740] usb 2-1.2: 88: ff ff ff ff ff ff ff ff
[16064.324744] usb 2-1.2: 90: ff ff ff ff ff ff ff ff
[16064.324747] usb 2-1.2: 98: ff ff ff ff ff ff ff ff
[16064.324750] usb 2-1.2: a0: ff ff ff ff ff ff ff ff
[16064.324754] usb 2-1.2: a8: ff ff ff ff ff ff ff ff
[16064.324757] usb 2-1.2: b0: ff ff ff ff ff ff ff ff
[16064.324761] usb 2-1.2: b8: 20 20 23 00 00 00 ff ff
[16064.324764] usb 2-1.2: c0: ff 01 20 13 00 00 00 ff
[16064.324767] usb 2-1.2: c8: 00 00 ff ff ff ff ff ff
[16064.324771] usb 2-1.2: d0: da 0b 8b 81 e7 47 02 00
[16064.324774] usb 2-1.2: d8: e0 4c 05 78 d3 09 03 52
[16064.324777] usb 2-1.2: e0: 65 61 6c 74 65 6b 0d 03
[16064.324781] usb 2-1.2: e8: 38 30 32 2e 31 31 6e 20
[16064.324784] usb 2-1.2: f0: 4e 49 43 00 ff ff ff ff
[16064.324788] usb 2-1.2: f8: ff ff ff ff ff ff ff ff
[16064.324791] usb 2-1.2: 100: ff ff ff ff ff ff ff ff
[16064.324795] usb 2-1.2: 108: ff ff ff ff ff ff ff ff
[16064.324798] usb 2-1.2: 110: ff ff ff ff ff ff ff 0d
[16064.324801] usb 2-1.2: 118: 03 00 05 00 30 00 00 00
[16064.324805] usb 2-1.2: 120: 00 93 ff ff ff ff ff ff
[16064.324808] usb 2-1.2: 128: ff ff ff ff ff ff ff ff
[16064.324812] usb 2-1.2: 130: f6 a8 98 2d 03 92 98 00
[16064.324815] usb 2-1.2: 138: fc 8c 00 11 9b 44 02 0a
[16064.324818] usb 2-1.2: 140: ff ff ff ff ff ff ff ff
[16064.324822] usb 2-1.2: 148: ff ff ff ff ff ff ff ff
[16064.324825] usb 2-1.2: 150: ff ff ff ff ff ff ff ff
[16064.324829] usb 2-1.2: 158: ff ff ff ff ff ff ff ff
[16064.324832] usb 2-1.2: 160: ff ff ff ff ff ff ff ff
[16064.324835] usb 2-1.2: 168: ff ff ff ff ff ff ff ff
[16064.324839] usb 2-1.2: 170: ff ff ff ff ff ff ff ff
[16064.324842] usb 2-1.2: 178: ff ff ff ff ff ff ff ff
[16064.324846] usb 2-1.2: 180: ff ff ff ff ff ff ff ff
[16064.324849] usb 2-1.2: 188: ff ff ff ff ff ff ff ff
[16064.324853] usb 2-1.2: 190: ff ff ff ff ff ff ff ff
[16064.324856] usb 2-1.2: 198: ff ff ff ff ff ff ff ff
[16064.324860] usb 2-1.2: 1a0: ff ff ff ff ff ff ff ff
[16064.324863] usb 2-1.2: 1a8: ff ff ff ff ff ff ff ff
[16064.324867] usb 2-1.2: 1b0: ff ff ff ff ff ff ff ff
[16064.324870] usb 2-1.2: 1b8: ff ff ff ff ff ff ff ff
[16064.324874] usb 2-1.2: 1c0: ff ff ff ff ff ff ff ff
[16064.324877] usb 2-1.2: 1c8: ff ff ff ff ff ff ff ff
[16064.324880] usb 2-1.2: 1d0: ff ff ff ff ff ff ff ff
[16064.324884] usb 2-1.2: 1d8: ff ff ff ff ff ff ff ff
[16064.324887] usb 2-1.2: 1e0: ff ff ff ff ff ff ff ff
[16064.324891] usb 2-1.2: 1e8: ff ff ff ff ff ff ff ff
[16064.324894] usb 2-1.2: 1f0: ff ff ff ff ff ff ff ff
[16064.324898] usb 2-1.2: 1f8: ff ff ff ff ff ff ff ff
[16064.324905] usb 2-1.2: RTL8192EU rev B (SMIC) 2T2R, TX queues 3, WiFi=1, BT=0, GPS=0, HI PA=0
[16064.324909] usb 2-1.2: RTL8192EU MAC: 00:e0:4c:05:78:d3
[16064.324913] usb 2-1.2: rtl8xxxu: Loading firmware rtlwifi/rtl8192eu_nic.bin
[16064.325153] usb 2-1.2: Firmware revision 19.0 (signature 0x92e1)
[16065.436610] rtl8xxxu 2-1.2:1.0 wlx00e04c0578d3: renamed from wlan0
[16065.481591] IPv6: ADDRCONF(NETDEV_UP): wlx00e04c0578d3: link is not ready
```

### What actually happened?

When I connect the same device on ev3dev stretch I see the following

```
[  215.695444] usb 1-1: new full-speed USB device number 2 using ohci-da8xx
[  215.973081] usb 1-1: New USB device found, idVendor=0bda, idProduct=818b
[  215.973145] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  215.973180] usb 1-1: Product: 802.11n NIC
[  215.973212] usb 1-1: Manufacturer: Realtek
[  215.973243] usb 1-1: SerialNumber: 00e04c000001
```
Looking at the `firmware-realtek` package in stretch, it does appear to contain `rtlwifi/rtl8192eu_nic.bin`: https://packages.debian.org/stretch/firmware-realtek but for some reason it is not being loaded. So maybe it's a matter of the right `modprobe` incantations?

It appears this devices is supported by the `rtl8xxxu` driver, and on my laptop `modinfo rtl8xxxu` returns a lot of stuff, while on the EV3 it returns `modinfo: ERROR: Module rtl8xxxu not found.`

### What hardware and software are you using (including version numbers)?

```
Image file:         ev3dev-stretch-ev3-generic-2018-06-13
Kernel version:     4.14.49-ev3dev-2.0.2-ev3
Brickman:           0.10.0
ev3devKit:          0.5.2
Board:              board0
BOARD_INFO_HW_REV=7
BOARD_INFO_MODEL=LEGO MINDSTORMS EV3
BOARD_INFO_ROM_REV=6
BOARD_INFO_SERIAL_NUM=00165335A852
BOARD_INFO_TYPE=main
```

**Comment by dlech** [comment]
we are currently using the same rtl8192cu driver as Raspberry Pi instead of this driver. The reason being that some people have reported that it has better performance. If we enable the rtl8xxxu driver, then both drivers will try to load.

So either we need to drop the out of tree driver that we are currently using or we need to ship ev3dev with a modules blacklist entry for one of the drivers if we enable both.

**Comment by pepijndevos** [comment]
It seems that's what Raspberry Pi is doing, ship the out of tree version, which has better perf for some users, but blacklist the in-tree one. So users that are not supported by the out of tree one can easily change the blacklist.

Currently the upstream driver is not shipped at all, right? So there is no easy way for me to just enable it.

**Comment by dlech** [comment]
you would have to compile your own kernel to enable it.

**Comment by pepijndevos** [comment]
Closing this now. Thanks for the merge! Can I expect to see these changes anytime soon in the docker images, or are these packages not updated that often?

**Comment by dlech** [comment]
the ev3-config package is already available, so you can `apt update && apt upgrade` it. kernel releases are rather time consuming though, so not sure when the next one will be.

**Comment by dlech** [comment]
Fix released in `ev3dev-2.1.0` kernel.
